### PR TITLE
Switch to pyproject.toml

### PR DESCRIPTION
**Pull request by quaxsze**
Fix https://github.com/etalab/data.gouv.fr/issues/939

### CODE REVIEW

**Review by maudetes — COMMENTED** [review]
I may miss something but it seems that we're not specifying requirements in pyproject.toml (we had `install_requires`), thus having only `udata` being install and packaged instead of udata + its dependencies

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    "vine==1.3.0",
+    "voluptuous==0.11.7",
+    "werkzeug==0.16.1",
+    "wtforms-json==0.3.3",
+    "wtforms==2.2.1",
+    "xmltodict==0.12.0"
+]
+
+[project.optional-dependencies]
+dev = [
```
Comment (maudetes) [inline pyproject.toml:95]: We should probably document how to install these? Or should we install based on requirements/develop.pip ?

**Review by maudetes — COMMENTED** [review]
Nice!

I have some issues on some deps when running `inv serve`. In particular, I saw mistakes on:
- MarkupSafe
- kombu[redis]

**Review by maudetes — COMMENTED** [review]
I think we're all good on udata side! :clap:

**Comment by ThibaudDauce** [comment]
Closing for now since the issue was closed

**Pull request closed without merging**